sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/ListingFileCatalog.scala

```diff
@@ -64,11 +66,18 @@ class ListingFileCatalog(
   }
 
   override def refresh(): Unit = {
+    refresh0(true)
+  }
+
+  private def refresh0(invalidateSharedCache: Boolean): Unit = {
```

Comment:
It's shared within a single table. Confusingly, you can have views over the table that are cached in CacheManager, so we need this to invalidate the table cache as well on refresh.